ftfy/__init__.py

```diff
@@ -6,211 +6,538 @@
+    or in the full documentation at ftfy.readthedocs.org.
+
+    For convenience and backward compatibility, the configuration can also
+    take the form of keyword arguments, which will set the equivalently-named
+    fields of the TextFixerConfig object.
+
+    For example, here are two ways to fix text but skip the "uncurl_quotes"
+    step::
+
+        fix_text(text, TextFixerConfig(uncurl_quotes=False))
+        fix_text(text, uncurl_quotes=False)
+
+    This function fixes text in independent segments, which are usually lines
+    of text, or arbitrarily broken up every 1 million codepoints (configurable
+    with `config.max_decode_length`) if there aren't enough line breaks. The
+    bound on segment lengths helps to avoid unbounded slowdowns.
+
+    ftfy can also provide an 'explanation', a list of transformations it applied
+    to the text that would fix more text like it. This function doesn't provide
+    explanations (because there may be different fixes for different segments
+    of text).
+
+    To get an explanation, use the :func:`fix_and_explain()` function, which
+    fixes the string in one segment and explains what it fixed.
     """
+
+    if config is None:
+        config = TextFixerConfig()
+    config = config._replace(**kwargs)
+    config = config._replace(explain=False)
     if isinstance(text, bytes):
-        raise UnicodeError(fixes.BYTES_ERROR_TEXT)
+        raise UnicodeError(BYTES_ERROR_TEXT)
 
     out = []
     pos = 0
     while pos < len(text):
-        textbreak = text.find('\n', pos) + 1
-        fix_encoding_this_time = fix_encoding
+        textbreak = text.find("\n", pos) + 1
         if textbreak == 0:
             textbreak = len(text)
-        if (textbreak - pos) > max_decode_length:
-            fix_encoding_this_time = False
-
-        substring = text[pos:textbreak]
-
-        if fix_entities == 'auto' and '<' in substring and '>' in substring:
-            # we see angle brackets together; this could be HTML
-            fix_entities = False
-
-        out.append(
-            fix_text_segment(
-                substring,
-                fix_entities=fix_entities,
-                remove_terminal_escapes=remove_terminal_escapes,
-                fix_encoding=fix_encoding_this_time,
-                uncurl_quotes=uncurl_quotes,
-                fix_latin_ligatures=fix_latin_ligatures,
-                fix_character_width=fix_character_width,
-                fix_line_breaks=fix_line_breaks,
-                fix_surrogates=fix_surrogates,
-                remove_control_chars=remove_control_chars,
-                remove_bom=remove_bom,
-                normalization=normalization,
-            )
-        )
+        if (textbreak - pos) > config.max_decode_length:
+            textbreak = pos + config.max_decode_length
+
+        segment = text[pos:textbreak]
+        if config.unescape_html == "auto" and "<" in segment:
+            config = config._replace(unescape_html=False)
```

Comment:
On the first point, that's intentional. Consider an HTML document whose second line is `<html>` and whose 10th line contains `&gt;`. We don't want to replace `&gt;` because it appears to be in an HTML document.

It's important to do it in both places because:

- `fix_and_explain` can be called directly, and needs to know about the angle bracket thing even if `fix_text` isn't being run
- `fix_text` can remember it's seen an angle bracket, so it knows to call `fix_and_explain` with `unescape_html=False` on later lines, so that `fix_and_explain` doesn't have to see an angle bracket on every single line

Reply:
I can totally buy that.